src/screens/home/searchProduct.js

```diff
@@ -0,0 +1,115 @@
+import {View, FlatList, StyleSheet} from "react-native"
+import React, {useEffect, useRef, useState, useCallback} from "react"
+import Color from "@common/Color"
+import {Header, Icon} from "react-native-elements"
+import SearchBar from "@components/searchbar"
+import {useIsFocused, useNavigation} from "@react-navigation/native"
+import Icons from "@common/Icon"
+import {homePage} from "@common/navigator"
+import {useSelector, useDispatch} from "react-redux"
+import {getProductSearch} from "@redux/slices/product"
+import _ from "lodash"
+import SearchItem from "./containers/searchItem/historyItem"
+import Loading from "../../components/loading"
+
+const SearchProduct = () => {
+  const products = useSelector((state) => state.product.searchProduct.data)
+
+  const [search, setSearch] = useState("")
+  const [loading, setLoading] = useState(true)
+  const dispatch = useDispatch()
+  const navigation = useNavigation()
+  const isFocused = useIsFocused()
+  const textInput = useRef(null)
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const sendSearchersWord = useCallback(
+    _.debounce(
+      (val) =>
+        dispatch(getProductSearch({page: 1, name: val}))
+          .unwrap()
+          .then((result) => {
+            if (result) {
+              setLoading(false)
+            }
+          }),
+      500,
+    ),
```

Comment:
Why don't use loading flag in redux slice instead of creating loading state in every components?